### ISSUE TITLE
Connectivity: `getAllDestinationProperties()` no full replacement for deprecated `tryGetAllDestinations(options)`

### ISSUE THREAD

**Issue by LukasHeimann** [SAP/cloud-sdk-java#558]
## Issue Description
https://github.com/SAP/cloud-sdk-java/pull/199 deprecated some old destination list retrieval APIs in favor of `getAllDestinationProperties()` on the destination service. However, the old method allowed to pass in DestinationOptions, while the new version doesn't.

In our code, we want to make sure to now accidentally respond with the list of destinations of the provider tenant when requests should be happening in the subscriber tenants only. In the past, this was possible with:

```java
  private final DestinationService service = new DestinationService();

  private static final DestinationOptions options = DestinationOptions
      .builder()
      .augmentBuilder(
          DestinationServiceOptionsAugmenter
              .augmenter()
              .retrievalStrategy(DestinationServiceRetrievalStrategy.ONLY_SUBSCRIBER))
      .build();

  public List<String> getAllDestinationsDeprecated() {
    var iterable = this.service.tryGetAllDestinations(options).get();
    var stream = StreamSupport.stream(iterable.spliterator(), false);
    return stream
        .filter(d -> DestinationType.HTTP.equals(d.get(DestinationProperty.TYPE).getOrNull()))
        .filter(d -> d.get(DestinationProperty.NAME).isDefined())
        .map(d -> d.get(DestinationProperty.NAME).get())
        .toList();
  }
```

The behavior is as expected -- go in as the provider tenant, and the request fails.

The new API is to be used in a similar fashion, it just doesn't fail when in the context of the provider tenant:

```java
  public List<String> getAllDestinationsNew() {
    return this.service.getAllDestinationProperties().stream()
        .filter(d -> DestinationType.HTTP.equals(d.get(DestinationProperty.TYPE).getOrNull()))
        .filter(d -> d.get(DestinationProperty.NAME).isDefined())
        .map(d -> d.get(DestinationProperty.NAME).get())
        .toList();
  }
```

Ideally, the new API `getAllDestinationProperties()` would also allow to receive `DestinationOptions`.

### Impact / Priority

Affected development phase: Production (degraded second line of defense for productive deployments)
 
Impact: Inconvenience

Timeline: Go-live planned for this year

## Project Details

* SDK Version: 5.11.0
* Link to GitHub repo: n/a
* Project type, for example:
    * [ ] CAP Project
    * [ ] SDK Maven Archetype
    * [x] None of the above:
* Platform:
    * [x] Cloud Foundry
    * [ ] Deploy with Confidence (Cloud Foundry)
    * [ ] None of the above:

-----

### Checklist

- [x] Checked out the [documentation](https://sap.github.io/cloud-sdk/docs/java/sdk-java-troubleshooting-frequent-problems) and [Stack Overflow](https://stackoverflow.com/questions/tagged/sap-cloud-sdk)
- [x] Description provided with all relevant information
- [ ] Exception and stack trace provided
- [ ] Attached debug logs
- [ ] Attached dependency tree
- [x] Provided Cloud SDK version & link to relevant source code

**Comment by KavithaSiva** [comment]
Hi @LukasHeimann,

Thanks for reaching out to us.
We will try to provide this feature in our upcoming release.

Regards,
Kavitha

**Comment by MatKuhr** [comment]
Hi @LukasHeimann, glad to let you know the improvement has been implemented and [shipped with version 5.12.0](https://sap.github.io/cloud-sdk/docs/java/release-notes#5120---september-03-2024). Please let us know if something doesn't work as expected 😉

**Comment by LukasHeimann** [comment]
Hi @MatKuhr -- really great, that was way faster than expected! Looks good and works like a breeze!
